include/container3d.h

```diff
@@ -52,8 +52,13 @@ class Container3D{
     T& getElement(const std::array<int,3> coord){
       return _data[coord[2] * _n_elements[0] * _n_elements[1] + coord[1] * _n_elements[0] + coord[0]];}
     // get boundaries
-    std::array<unsigned long int,3> getNumElements() const {
-      return _n_elements;
+    template <typename Q = unsigned long>

```

Comment:
Does it change something ?

Reply:
Yes, my changes make it so you can call the function and at the same time cast the array to a different type.

That means that when you want to get `_n_elements` not as `array<unsigned long,3>` but as `array<unsigned,3>` you can simply call the function as `getNumElements<unsigned>()`. Then you don't need to manually cast the elements afterwards.